### ISSUE TITLE
wxMaxima does not start: assert ""tlw"" failed

### ISSUE THREAD

**Issue by perryprog** [wxMaxima-developers/wxmaxima#1152]
When starting wxMaxima (macOS) with the latest head of wxwidgets I get this unrecoverable error:
```
❯ ./wxMaxima.app/Contents/MacOS/wxmaxima
12:00:12: Debug: Not setting language 0
12:00:12: Debug: Compatibility issue: Tab is not supported as a keyboard accelerator key on GTK
12:00:12: Debug: Compatibility issue: Tab is not supported as a keyboard accelerator key on GTK
12:00:12: Debug: Compatibility issue: Tab is not supported as a keyboard accelerator key on GTK
12:00:12: Debug: Compatibility issue: Tab is not supported as a keyboard accelerator key on GTK
./src/osx/window_osx.cpp(722): assert ""tlw"" failed in DoClientToScreen(): TopLevel window missing
```

It should be noted that in order to be compatible with the latest wxwidgets version I have this patch applied, but I doubt it's related as it's worked before this issue.
```patch
diff --git a/src/StatusBar.cpp b/src/StatusBar.cpp
index 0aa2fd40..6dafbb71 100644
--- a/src/StatusBar.cpp
+++ b/src/StatusBar.cpp
@@ -194,7 +194,7 @@ void StatusBar::NetworkStatus(networkState status)
     case receive:
     {
       ReceiveTimer.StartOnce(200);
-      wxTimerEvent dummy;
+      wxTimerEvent dummy = *new wxTimerEvent(*new wxTimer());
       OnTimerEvent(dummy);
       if((m_oldmaximaPercentage >= 0) &&(m_maximaPercentage < 0))
         m_networkStatus->SetToolTip(m_stdToolTip);
@@ -203,7 +203,7 @@ void StatusBar::NetworkStatus(networkState status)
     case transmit:
     {
       SendTimer.StartOnce(200);
-      wxTimerEvent dummy;
+      wxTimerEvent dummy = *new wxTimerEvent(*new wxTimer());
       OnTimerEvent(dummy);
       if((m_oldmaximaPercentage >= 0) &&(m_maximaPercentage < 0))
         m_networkStatus->SetToolTip(m_stdToolTip);
diff --git a/src/Worksheet.cpp b/src/Worksheet.cpp
index f13fdc57..9e564e3a 100644
--- a/src/Worksheet.cpp
+++ b/src/Worksheet.cpp
@@ -7872,7 +7872,7 @@ void Worksheet::OnSetFocus(wxFocusEvent &event)

   // And we want the cursor start in its visible phase.

-  wxTimerEvent dummy;
+  wxTimerEvent dummy = *new wxTimerEvent(*new wxTimer());
   dummy.SetId(CARET_TIMER_ID);
   OnTimer(dummy);
   event.Skip();
```

**Comment by gunterkoenigsmann** [comment]
Mac testers are always welcome!

Do you have a method to get a backport somehow? (I hope the assert window somehow can produce one)

**Comment by perryprog** [comment]
Oh right, sure. For some reason I can't copy and paste it, hopefully a screenshot will do.
<img width="976" alt="image" src="https://user-images.githubusercontent.com/12972285/63958888-079adb00-ca59-11e9-8201-e91081eb45a2.png">

**Comment by gunterkoenigsmann** [comment]
Wow... ...the "variables" sidebar is new => the error could not have been triggered in prior versions of wxMaxima. But it happens in the constructor of an innocent wxGrid that is part of the main window => I cannot guaranteed if that can be resolved in wxMaxima or if wxWidgets is at fault.

**Comment by gunterkoenigsmann** [comment]
But you seem to be using dark mode...
...Should the worksheet background be black in dark mode? What do you think? I don't own a mac and therefore lack any experience with it.

**Comment by gunterkoenigsmann** [comment]
Do you use multiple screens at the same time?

---

I have added your patch upstreâm. Hope you allow me to do so.

**Comment by perryprog** [comment]
The worksheet shouldn't be black in dark mode. The integration with dark mode is better than most apps thanks to WxWidgets. I only use one screen currently.

Thanks for adding in the patch!

**Comment by perryprog** [comment]
The change you made to the patch causes a compilation error.

```c++
../src/StatusBar.cpp:197:20: error: no viable conversion from 'wxTimerEvent *' to 'wxTimerEvent'
      wxTimerEvent dummy = new wxTimerEvent(*new wxTimer());
                   ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/wx-3.1/wx/timer.h:159:24: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'wxTimerEvent *' to 'const wxTimerEvent &' for 1st argument; dereference the argument with *
class WXDLLIMPEXP_BASE wxTimerEvent : public wxEvent
                       ^
/usr/local/include/wx-3.1/wx/timer.h:162:5: note: candidate constructor not viable: no known conversion from 'wxTimerEvent *' to 'wxTimer &' for 1st argument
    wxTimerEvent(wxTimer& timer)
    ^
../src/StatusBar.cpp:206:20: error: no viable conversion from 'wxTimerEvent *' to 'wxTimerEvent'
      wxTimerEvent dummy = new wxTimerEvent(*new wxTimer());
                   ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/wx-3.1/wx/timer.h:159:24: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'wxTimerEvent *' to 'const wxTimerEvent &' for 1st argument; dereference the argument with *
class WXDLLIMPEXP_BASE wxTimerEvent : public wxEvent
                       ^
/usr/local/include/wx-3.1/wx/timer.h:162:5: note: candidate constructor not viable: no known conversion from 'wxTimerEvent *' to 'wxTimer &' for 1st argument
    wxTimerEvent(wxTimer& timer)
    ^
2 errors generated.
```

I'm not sure why (I'm no c++ whizz) but when I added in the change I had to do a double dereference, like so: `*new wxTimerEvent(*new wxTimer())` I'm not sure if there's a way around that.

**Comment by gunterkoenigsmann** [comment]
Gave it a 2nd try. Let's hope it works now.

**Comment by perryprog** [comment]
That fixed the timer issue. Thanks again.

**Comment by perryprog** [comment]
Right, this must be a breaking change or a bug in WxWidgets since going back over old commits I still get the same issue. Feel free to close if you want.

**Comment by gunterkoenigsmann** [comment]
As the error will sooner or later hit a wxWidgets release (and therefore all users) if we don't prevent this I would ask you two favours:

1. Would you be willing to test if the "grid" sample that comes with wxWidgets is affected, too?
2. And if it is: Would you be willing to report this on https://trac.wxwidgets.org/?

Thanks a lot,
and kind regards,

  Gunter.

**Comment by perryprog** [comment]
Yeah, it's a wxWidgets issue. Filing a bug now.

**Comment by perryprog** [comment]
[All done.](https://trac.wxwidgets.org/ticket/18481)

**Comment by gunterkoenigsmann** [comment]
Cool! Thanks a lot!

**Comment by perryprog** [comment]
Should be closed with [#1527](https://github.com/wxWidgets/wxWidgets/pull/1527).
